improver/developer_tools/metadata_interpreter.py

```diff
@@ -0,0 +1,578 @@
+                try:
+                    perc_coord = find_percentile_coordinate(cube)
+                except CoordinateNotFoundError:
+                    coords = get_coord_names(cube)
+                    if any(
+                        [cube.coord(coord).var_name == "threshold" for coord in coords]
+                    ):
+                        self.field_type = self.PROB
+                        self.check_probability_cube_metadata(cube)
+                    else:
+                        self.field_type = self.DIAG
+                else:
+                    self.field_type = self.PERC
+                    if perc_coord.name() != PERC_COORD:
+                        self.errors.append(
+                            f"Percentile coordinate should have name {PERC_COORD}, "
+                            f"has {perc_coord.name()}"
+                        )
+
+                    if perc_coord.units != "%":
+                        self.errors.append(
+                            "Percentile coordinate should have units of %, "
+                            f"has {perc_coord.units}"
+                        )
+
+            self.check_cell_methods(cube)
+
+        # 2) Interpret model and blend information from cube attributes
+        self.check_attributes(cube.attributes)
+
+        # 3) Check whether expected coordinates are present
+        coords = get_coord_names(cube)
+        if "spot_index" in coords:
+            self.check_spot_data(cube, coords)
+
+        if self.field_type == self.ANCIL:
+            # there is no clear standard for time coordinates on static ancillaries
+            pass
+        elif (
+            cube.coords("time")
+            and len(cube.coord_dims("time")) == 2
+            and not self.blended
+        ):
+            # 2D time coordinates are only present on global day-max diagnostics that
+            # use a local time zone coordinate. These do not have a 2D forecast period.
+            expected_coords = set(LOCAL_TIME_COORDS + UNBLENDED_TIME_COORDS)
+            expected_coords.discard("forecast_period")
+            self._check_coords_present(coords, expected_coords)
+        elif self.blended:
+            self._check_coords_present(coords, BLENDED_TIME_COORDS)
+        else:
+            self._check_coords_present(coords, UNBLENDED_TIME_COORDS)
+
+        # 4) Check points are equal to upper bounds for bounded time coordinates
+        for coord in ["time", "forecast_period"]:
+            if coord in get_coord_names(cube):
+                self._check_coord_bounds(cube, coord)
+
+        # 5) Check datatypes on data and coordinates
+        try:
+            check_mandatory_standards(cube)
+        except ValueError as cause:
+            self.errors.append(str(cause))
+
+        # 6) Raise collated errors if present
+        if self.errors:
+            raise ValueError("\n".join(self.errors))
+
+
+def display_interpretation(interpreter, verbose=False):
```

Comment:
Why have you chosen to make this a separate function rather than a method on the Interpreter class?

Reply:
Because the interpreter class was getting huge, and I thought it was doing too much.  I decided it would make sense to separate this into two concerns:

1. Interpreting what was in the file, including raising errors if that was impossible (ie if the metadata were incomplete or inconsistent)
2. Translating the interpretation into a simple, human-readable output - essentially parsing strings - which was making the interpreter very verbose when done inline

I felt like populating the interpreter was a self-contained activity that fitted well into a plugin, and parsing the output could fit separately.

Reply:
Fair enough